includes/discovery/functions.inc.php

```diff
@@ -968,7 +968,15 @@ function discovery_process(&$valid, $os, $sensor_class, $pre_cache)
                         $value = $user_function($value);
                     }
 
-                    $uindex = str_replace('{{ $index }}', $index, isset($data['index']) ? $data['index'] : $index);
+                    $uindex = $index;
+                    if (isset($data['index'])) {
+                        if (Str::contains($data['index'], '{{')) {
+                            $uindex = trim(YamlDiscovery::replaceValues('index', $index, null, $data, $pre_cache));
+                        } else {
+                            $uindex = str_replace('{{ $index }}', $index, $data['index']);
+                        }
+                    }
+
```

Comment:
```suggestion
                    $uindex = isset($data['index']) 
                        ? YamlDiscovery::replaceValues('index', $index, null, $data, $pre_cache) 
                        : $index;

```

replaceValues will cover the {{ $index }} too.

Reply:
And also, the str_replace would never replace anything anyway because of the Str::contains check.

Reply:
We cannot use replaceValues alone because replace value will also match an OID without `{{ $` and ` }}`. Which breaks a lot of indexes. We must avoid to call replaceValues in that case. 

The cases are : 
- uindex defaults to $index.
- if we have an index defined in the YAML
  - if it is safe to use replaceValues on it, then we do it
  - if not we use this YAML defined index without any replacement

so the code could be simplified to : 
```
                    $uindex = $index;
                    if (isset($data['index'])) {
                        if (Str::contains($data['index'], '{{')) {
                            $uindex = trim(YamlDiscovery::replaceValues('index', $index, null, $data, $pre_cache));
                        } else {
                            $uindex = $data['index'];
                        }
                    }
```

Reply:
I forgot that does literal swapping, LGTM